cluster-api/pkg/apis/cluster/v1alpha1/cluster_types_test.go

```diff
@@ -85,3 +86,67 @@ func crudAccessToClusterClient(t *testing.T, cs *clientset.Clientset) {
 		t.Fatalf("Number of items in Items list should be 0, but is %d.", itemLength)
 	}
 }
+
+func clusterValidationTest(t *testing.T, cs *clientset.Clientset) {
+	tests := []struct {
+		name        string
+		cluster     v1alpha1.Cluster
+		errExpected bool
+	}{
+		{
```

Comment:
Can we get a test case that doesn't error?

Reply:
Added. 

In terms of defaulting, i think defaulting happen before validation. If we decide to add defaulting in the future, we can modify the validation logic/test here as desired.